icu4c/source/test/cintltst/ucptrietest.c

```diff
@@ -705,6 +707,61 @@ testBuilder(const char *testName, const UMutableCPTrie *mutableTrie,
     testTrieGetRanges(testName, NULL, mutableTrie, UCPMAP_RANGE_NORMAL, 0, checkRanges, countCheckRanges);
 }
 
+static void
+trieTestGolden(const char *testName,
+             const UCPTrie* trie,
+             const CheckRange checkRanges[],
+             int32_t countCheckRanges) {
+    log_verbose("golden testing Trie '%s'\n", testName);
+
+    UErrorCode status = U_ZERO_ERROR;
+    const char *testdatapath = loadSourceTestData(&status);
+    char goldendatapath[512];
+    // note: snprintf always writes a NUL terminator.
+    snprintf(goldendatapath, 512, "%scodepointtrie%s%s.toml",
+        testdatapath, U_FILE_SEP_STRING, testName);
+
+    // Write the data into a tmpfile (memstream is not portable)
+    FILE* stream = tmpfile();
+    usrc_writeCopyrightHeader(stream, "#", 2021);
+    fputs("#\n# machine-generated by: ucptrietest.c\n", stream);
```

Comment:
Please call usrc_writeFileNameGeneratedBy() for consistency.

(nice refactoring, by the way)

Reply:
done.